### DISCUSSION TITLE
Images disappearing from products seemingly randomly

### DISCUSSION THREAD

**Discussion by vettloffah** [swellstores/community#482] [Help] [answered]
We have been having an ongoing issue, where product images are "disappearing" from products. Basically, we'll notice that a product is missing one or more images that previously had been loaded and displayed properly. It does not seem to be caused by any specific action so we cannot reproduce the issue. It's hard to troubleshoot, since we aren't even sure when it happens. We just notice an image is missing and we don't know for how long it's been missing. This has been ongoing for months and has affected probably 10 products over time. Sometimes it happens to products that we haven't updated in many months through the dashboard. We have one webhook that updates stock levels, but that only passes a `post` request to `products:stock` and doesn't update the product data itself.

I've tried working with support, but since we can't reproduce the issue intentionally it seems like they gave up investigating it.

I'm wondering if anyone else has experienced this and might be able to offer some advice, it's really starting to drive my staff crazy.

**Comment by ericingram** [comment]
We've heard about issues like this a couple times. Once we found that the user's frontend was modifying the product. Another time it happened when someone was deleting a product option, which then deletes related variants and their images, and adding the option back with the expectation that the images would be restored. Just to say that these are factors we've seen before.

Have you tried looking at `events` for the product to see when the images were removed? And then looking at logs in the developer console to see where the API call came from. Note: if the call originated from the dashboard itself then it wouldn't show up in the developer console logs, but that's another hint too.

In any case I'll talk to the team myself and get back to you.

**Comment by vettloffah** [comment]
How would the query look to get back all events for a specific product? I'm having trouble getting it to return any results.

**Reply by ericingram** [reply]
It should be like this:
```
GET /events?data.id={product_id}
```

**Comment by logeshswell** [comment, marked as answer]
@vettloffah We noticed a few issues in our CDN and deployed the fix for it.
Can you try deleting the previous image and re-upload it back?
